src/core-taggeds-failure/Failure/Failure.T/Failure.T.ToString.cs

```diff
@@ -6,6 +7,23 @@ partial struct Failure<TFailureCode>
 {
     public override string ToString()
         =>
-        Invariant(
-            $"Failure<{typeof(TFailureCode).Name}>:{{ \"FailureCode\": {FailureCode}, \"FailureMessage\": \"{FailureMessage}\" }}");
+        new StringBuilder(
```

Comment:
Since `string.Format` (which is implicitly used by the string interpolation) implementation uses special [ValueStringBuilder](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Text/ValueStringBuilder.cs) `ref struct` and other optimizations (like `stackalloc`, etc.), explicit using the `StringBuilder` looks a kind of premature optimization when building the `ToString` result.

So, consider to keep using the `Invariant` with string interpolation by just adding handling the new parameter (`SourceException`), like this:
```cs
    public override string ToString()
        =>
        Invariant(
            $"Failure<{typeof(TFailureCode).Name}>:{{ \"FailureCode\": {FailureCode}, \"FailureMessage\": \"{FailureMessage}\", \"SourceException\": {InnerSourceExceptionString()}}}");

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private string InnerSourceExceptionString()
        =>
        SourceException is null ? "null" : Invariant($"\"{SourceException}\"");
```